src/main/kotlin/com/refinedmods/refinedstorage/gui/screen/BaseScreen.kt

```diff
@@ -0,0 +1,340 @@
+                slot.isSizeAllowed
+
+        if(!valid)
+            return super.onMouseClick(slot, slotId, mouseButton, type)
+
+        when(slot) {
+
+            is FilterSlot -> {
+                if (!slot.getStack().isEmpty) {
+                    return when {
+                        slot.isAlternativesAllowed && hasControlDown() ->
+                            openAlternateScreen(player, slot)
+                        else ->
+                            openItemAmountScreen(player, slot)
+                    }
+                }
+            }
+
+            is FluidFilterSlot -> {
+                val fluidStack = slot.fluidInventory.getFluid(slot.id)
+                if (!fluidStack.isEmpty) {
+                    return when {
+                        slot.isAlternativesAllowed && hasControlDown() ->
+                            openAlternateScreen(player, slot, fluidStack)
+                        else ->
+                            openFluidAmountScreen(player, slot, fluidStack)
+                    }
+                }
+            }
+        }
+
+        super.onMouseClick(slot, slotId, mouseButton, type)
+    }
+
+    fun renderItem(
+        matrixStack: MatrixStack,
+        x: Int,
+        y: Int,
+        stack: ItemStack,
+        overlay: Boolean = false,
+        text: String? = null,
+        textColor: Int = 0
+    ) {
+        zOffset = Z_LEVEL_ITEMS
+        itemRenderer.zOffset = Z_LEVEL_ITEMS.toFloat()
+        itemRenderer.renderInGui(stack, x, y)
+        if (overlay) {
+            itemRenderer.renderGuiItemOverlay(textRenderer, stack, x, y, "")
+        }
+
+        zOffset = 0
+        itemRenderer.zOffset = 0f
+
+        text?.let { renderQuantity(matrixStack, x, y, it, textColor) }
+    }
+
+    fun addCheckBox(x: Int, y: Int, text: Text, checked: Boolean, onPress: Consumer<CheckboxWidget>): CheckboxWidget {
+        val checkBox = CheckboxWidget(x, y, text, checked, onPress)
+        this.addButton(checkBox)
+        return checkBox
+    }
+
+    fun addButton(
+        x: Int,
+        y: Int,
+        w: Int,
+        h: Int,
+        text: Text?,
+        enabled: Boolean,
+        visible: Boolean,
+        onPress: ButtonWidget.PressAction?
+    ): ButtonWidget  {
+        val button = ButtonWidget(x, y, w, h, text, onPress)
+        button.active
+        button.active = enabled
+        button.visible = visible
+        addButton(button)
+        return button
+    }
+
+    fun addSideButton(button: SideButton) {
+        button.x = this.x + -SideButton.WIDTH - 2
+        button.y = this.y + sideButtonY
+        sideButtonY += SideButton.HEIGHT + 2
+        addButton(button)
+    }
+
+    fun bindTexture(namespace: String, filenameInTexturesFolder: String) {
+        TrackedDataHandlerRegistry.FLOAT
+        MinecraftClient.getInstance().textureManager
+            .bindTexture(
+                TEXTURE_CACHE.computeIfAbsent("$namespace:$filenameInTexturesFolder")
```

Comment:
What is this for?

Reply:
*shurg* 
https://github.com/refinedmods/refinedstorage/blob/aba679a6b0f56f65bb9f14272815377db225df18/src/main/java/com/refinedmods/refinedstorage/screen/BaseScreen.java#L276``